src/relay.cpp

```diff
@@ -638,6 +638,10 @@ void callback(evutil_socket_t fd, short event, void *arg) {
     }
 
     auto udp_header = parse_udp(current_position, &tmp);
+    if ((current_position + ntohs(udp_header->len)) != ((uint8_t *)ptr + len)) {
+        syslog(LOG_WARNING, "UDP header length is invalid\n");
+        return;
+    }
```

Comment:
Agree this fix as current solution.  

In current code, we still not be able to handle malformed options (ip length, udp length are good, but wrong length in dhcpv6 options, like invalid length of TLV for option60), we don't care the options and just copy it and relay to server, this may leak risk to server.

In long term, maybe relay agent need to handle all kinds of malformed packets, like invalid option length (in current code, for dhcpv6 packets from client we just copy the packet, append required options behind packet). One solution is we generate a new packet (call response packet) based on walk of the options in DHCPv6 packets (call request packet) from client (or from DHCPv6 server), we go through request packets get all options, generate new response packet based on the content of request packets. During this process, we are be able to check all option definitions if malformed options we drop the packet and give warning, if wrong packet length (wrong udp length) we can also detected this in option loop. 

One good example is mdlayher/dhcp6.